ios/brave-ios/Sources/BraveWallet/Crypto/Stores/CryptoStore.swift

```diff
@@ -669,6 +669,9 @@ public class CryptoStore: ObservableObject, WalletObserverStore {
   func closeUserAssetsStore() {
     userAssetsStore?.tearDown()
     userAssetsStore = nil
+
+    // update `PortfolioStore` and `NFTStore` since user might have edited user assets
+    updateAssets()
```

Comment:
`PortfolioStore` & `NFTStore` both already observe `WalletUserAssetDataObserver` & call update in `userAssetUpdated()`. This just duplicates the call to update, and would call when assets aren't actually updated?

Reply:
Looks like the bug might be that `UserAssetsStore` isn't using `WalletUserAssetManagerType`'s `removeUserAsset` function and is deleting the asset manually?

Reply:
the bug is about `EditUserAssetsView` will be initiate again(`UserAssetsStore` won't initiate again since it's a reference from `CryptoStore`) since `PortfolioStore` is observing `userAssetUpdated` as well as `cachedBalanceRefreshed`. `WalletUserAssetManagerType`'s `removeUserAsset` will broadcasting multiple times since it's 1. updating user asset's visibility 2. remove cache balance 3. update user asset in `WalletService`. + initially `UserAssetsStore` is also observing `userAssetUpdated`, the `List` will be updated many times and cause race-condition. 
Essentially, this fix is 1. remove the asset (update database, remove cached balance, update `WalletService`) without broadcasting to all the `WalletUserAssetDataObserver` in this case `PortfolioStore` and `NFTStore` hence, the manual calling `updateAssets()` is needed when closing the `UserAssetsStore`.

Reply:
The crash error log I am seeing is indicating that the item isn't being removed from the collection view data source:
```
Invalid update: invalid number of items in section 0. The number of items contained in an existing section after the update (1708) must be equal to the number of items contained in that section before the update (1708), plus or minus the number of items inserted or deleted from that section (0 inserted, 1 deleted) and plus or minus the number of items moved into or out of that section (0 moved in, 0 moved out)
```
It seems like the item isn't being removed from the data source quick enough? Maybe due to the `Task` / async request?

I don't think we want to call update on PortfolioStore & NFTStore every single time the user opens & closes Edit User Assets, even when they haven't updated any asset.